metron-platform/metron-common/src/main/java/org/apache/metron/common/zookeeper/configurations/ProfilerUpdater.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.common.zookeeper.configurations;
+
+import static org.apache.metron.common.configuration.ConfigurationType.PROFILER;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.metron.common.configuration.ConfigurationType;
+import org.apache.metron.common.configuration.ConfigurationsUtils;
+import org.apache.metron.common.configuration.profiler.ProfilerConfig;
+import org.apache.metron.common.configuration.profiler.ProfilerConfigurations;
+import org.apache.metron.common.utils.JSONUtils;
+import org.apache.zookeeper.KeeperException;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class ProfilerUpdater extends ConfigurationsUpdater<ProfilerConfigurations> {
+  public ProfilerUpdater(Reloadable reloadable, Supplier<ProfilerConfigurations> configSupplier) {
+    super(reloadable, configSupplier);
+  }
+
+  @Override
+  public Class<ProfilerConfigurations> getConfigurationClass() {
+    return ProfilerConfigurations.class;
+  }
+
+  private ProfilerConfig readFromZookeeper(CuratorFramework client) throws Exception {
+    byte[] raw = client.getData().forPath(PROFILER.getZookeeperRoot());
+    return JSONUtils.INSTANCE.load(new ByteArrayInputStream(raw), ProfilerConfig.class);
+  }
+
+  @Override
+  public void forceUpdate(CuratorFramework client) {
+    try {
+      ConfigurationsUtils.updateConfigsFromZookeeper(getConfigurations(), client);
+    }
+    catch (KeeperException.NoNodeException nne) {
+      LOG.warn("No current global configs in zookeeper, but the cache should load lazily...");
+    }
+    catch(Exception e) {
+      LOG.warn("Unable to load global configs from zookeeper, but the cache should load lazily...", e);
+    }
+    try {
+      ProfilerConfig config = readFromZookeeper(client);
+      if(config != null) {
+        getConfigurations().updateProfilerConfig(config);
+      }
```

Comment:
The `ProfilerUpdater` is the only `ConfigurationsUpdater` implementation that has this additional logic in lines 61-73.  Looks fishy.  Why is it special?

Reply:
Hmm, looks-like just a difference in what is offered in `ConfigurationUtils`.  Seems necessary unless we were to unpack some of the logic from `ConfigurationUtils`.  But diving into `ConfigurationUtils` seems more effort than it is worth for this change.

Reply:
This is because the Profiler doesn't have a call in ConfigurationsUtils, unlike the rest of our fundamental types (e.g. `update{Parser,Enrichment,etc}ConfigsFromZookeeper`).  When calling those, an update to the global config is called and then the configs are updated.  This is done here and it's why it's a bit different.  I thought about moving this abstraction to `ConfigurationsUtils`, but I felt that class was already so onerous and needed to be fundamentally refactored so deeply that I didn't have the heart to create a `updateProfilerConfigsFromZookeeper` method.

Reply:
So to summarize, that was me answering my own question.  Ignore this. :)

Reply:
@cestella  Saw your response only after my last post.  Yes, makes complete sense.  Thanks